### ISSUE TITLE
Agent's log should not contain potentially security sensitive info

### ISSUE THREAD

**Issue by SergeyKleyman** [elastic/apm-agent-dotnet#515]
Example from https://discuss.elastic.co/t/apm-server-error-connection-refused/200109/3:
```
{{Scope}} Failed capturing request (failed to remove from ProcessingRequests) - This Span will be skipped in case it wasn't captured before. Request: method: "POST", URL: http://elastic:PleaseChangeMe@localhost:9200/_bulk
```

Agent logs URL including username and password: http://**elastic:PleaseChangeMe**@localhost:9200/_bulk

**Comment by gregkalapos** [comment]
This is kind of a duplicate of #202. Opened #568, which addresses also this problem.

**Comment by gregkalapos** [comment]
The specific log from the description is addressed in #568.

I personally would be fine with this - given this is planned for 7.5 and - and just in general during reviews make sure we don't log sensitive data - at least data that is not directly entered into some config by our users.

There was a comment here to maybe do more https://github.com/elastic/apm-agent-dotnet/pull/568#discussion_r339418480

I leave this open for discussion for a while and will bring this up during our next planning to see what others think.

**Comment by gregkalapos** [comment]
This was fixed in #568.

This was discussed in an agent weekly and no other agent does some magic filtering or so. It was agreed that on a default log level we should not log anything security sensitive, but if the log level is changed to a more detailed level it's fine to log security critical things, as this is something that the user actively triggers.

Closing this now, we can do another iteration later if we feel so.
